src/vaults/BalanceSheet.sol

```diff
@@ -325,4 +326,12 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
         queue.increase = 0;
         queue.decrease = 0;
     }
+
+    function _pricePoolPerAsset(PoolId poolId, ShareClassId scId) internal view returns (D18) {
+        return d18(TransientStorageLib.tloadUint128(keccak256(abi.encode("pricePoolPerAsset", poolId, scId))));
```

Comment:
CR

If price is 0 must default to pool manager price

Reply:
I wonder how we should handle when technically a price is actually 0.0.

Reply:
I asked myself the same thing yesterday and decided on just returning 0 then. This was influenced by the implementation of VaultPricingLib which defaulted to 0 when the price was 0. So if you had `price = 0 = pPAPS = pPPS / pPPA`, that also applied.

Reply:
So maybe the order of priority for pPPS is 

1. BS
2. SR = SyncRequests
3. PM

When the SR defaults to BS, this should be applied to the entire Vault Repository because SR is the SharePriceProvider. Though I dont remember if any contract still relies on that.

Reply:
I think we should never use the sync price anywhere as default besides in the sync one. The less implicit the action the easier. My main reason is not that it doesn't make sense. But it makes it harder to predict, simulate and reason about.

If somebody wants to have the sync price, they should overwrite the pool manager one before interacting with the bs.

Reply:
I also would make the pool manager the default sharePriceProvider.

Reply:
@hieronx and @wischli where is this provider actually used?